packages/metal-state/src/validators.js

```diff
@@ -132,9 +131,9 @@ const validators = {
 					required = validator.config.required;
 					validator = validator.config.validator;
 				}
-				if ((required && !isDefAndNotNull(value[key])) ||
-					isInvalid(validator(value[key]))) {
-					return composeError(ERROR_SHAPE_OF, name, context);
+				if ((required && !isDefAndNotNull(value)) ||
+					isInvalid(validator(value))) {
+					return validator(value, name + '.' + key, context);
```

Comment:
In cases of use with cascading objects, can return `undefined` values. Returns another value in `Expected` than defined. It has to be investigated.